lib/shiki-plugin.ts

```diff
@@ -45,27 +48,37 @@ function getDecorations({
       language = 'plaintext'
     }
 
-    const themeToApply = highlighter.getLoadedThemes().includes(theme)
-      ? theme
-      : highlighter.getLoadedThemes()[0]
+    const getThemeToApply = (theme: string): BundledTheme => {
+      if (highlighter.getLoadedThemes().includes(theme)) {
+        return theme as BundledTheme
+      } else {
+        return highlighter.getLoadedThemes()[0] as BundledTheme
+      }
+    }
 
-    const tokens = highlighter.codeToTokens(block.node.textContent, {
-      lang: language,
+    let options: CodeToTokensWithThemesOptions<BundledLanguage, BundledTheme>
 
-      // TODO: dual-theme should be optional. It should still support single theme
-      // if the tiptap extension isn't initialized with dual themes.
-      // theme: themeToApply,
+    if (themes) {
+
+      options = {
+        lang: language,
+        themes: {
+          light: getThemeToApply(lightTheme),
+          dark: getThemeToApply(darkTheme),
+        },
+      }
+    } else {
+      options = {
+        lang: language,
+        themes: {
+          light: getThemeToApply(theme),
+          dark: getThemeToApply(theme),
+        }
+      }
+    }
```

Comment:
This is going to force all users that had a single theme now to have both, you can see this happening if running the tests. This might be a breaking change for existing users.


I would suggest using `codeToTokensBase` for single themes for backward compatibility
```suggestion
    if (themes) {

      const tokens = highlighter.codeToTokens(block.node.textContent, {
        lang: language,
        themes: {
          light: getThemeToApply(lightTheme),
          dark: getThemeToApply(darkTheme),
        },
      })
    } else {
      const tokens = highlighter.codeToTokensBase(block.node.textContent, {
        lang: language,
        theme: getThemeToApply(theme),
      })
    }
```